sdl-core/src/main/scala/io/smartdatalake/util/hive/HiveUtil.scala

```diff
@@ -225,9 +224,10 @@ private[smartdatalake] object HiveUtil extends SmartDataLakeLogger {
 
     // check if partitionsOpt match with existing table definition
     if (tableExists) {
-      val configuredCols = partitions.toSet
-      val existingCols = getTablePartitionCols(table).getOrElse(Seq()).toSet
-      require( configuredCols==existingCols, s"(${table.fullName}) writeDfToHive: configured vs existing partition columns are different: configured=$configuredCols, existing=$existingCols" )
+      val existingPartitions = getTablePartitionCols(table).getOrElse(Seq())
+      val (partitionsMatch, configuredCols, existingCols) = SchemaUtil.checkPartitionMatch(partitions, existingPartitions, Environment.caseSensitive)
+
+      require( partitionsMatch, s"(${table.fullName}) writeDfToHive: configured vs existing partition columns are different: configured=$configuredCols, existing=$existingCols" )
```

Comment:
It would also be ok to use partitions and existingPartitions directly, maybe even clearer about the origin of different columns. WDYT?

Reply:
What do you mean? To put e.g. getTablePartitionCols(table)... directly as parameter into the checkPartitionMatch function? Or to remove the function as it's only used in this case here and do the logic directly here?

Reply:
I mean to simplify `SchemaUtil.checkPartitionMatch` so that it returns only partitionsMatch, and use $partitions and $existingPartitions in require text:
```suggestion
      require( partitionsMatch, s"(${table.fullName}) writeDfToHive: configured vs existing partition columns are different: configured=$partitions, existing=$existingPartitions" )
```

Reply:
As discussed, lets keep it like it is.